### ISSUE TITLE
Auto updates for search engines and device detection

### ISSUE THREAD

**Issue by mattkolb** [piwik/piwik#8980]
I noticed that a previous version started to automatically retrieve the latest version of the referrer spam list. Doing the same for search engines and device detection would be another thing that would be incredibly useful because some can't update the entire Piwik installation as fast as new devices come out and search engines change how they send referrer data.

**Comment by mattkolb** [comment]
Including this in the last 2.x release could also dramatically improve how well Piwik performs for those that can't update to 3.0

**Comment by sgiehl** [comment]
Good suggestion. We might easily be able to do that for search engines. I'll have a look

**Comment by sgiehl** [comment]
Autoupdate for search engines and social networks is now implemented. Device detection is a bit more complex...

**Comment by mattab** [comment]
@sgiehl we maybe don't need to auto update devices for now, it also gives a nice incentive for people to ugprade to latest piwik ;-)  

wondering, in case we change YML format in a future major version, does it mean that auto-update should only upgrade the same "major version" to prevent a future format breakage?

**Comment by sgiehl** [comment]
yes, changing the YML format might cause problems if we auto update without checking the version.
